src/main/java/org/skriptlang/skript/lang/comparator/Comparators.java

```diff
@@ -106,38 +109,60 @@ public static <T1, T2> void registerComparator(
-		if (first == null || second == null)
+		if (first == null || second == null) {
 			return Relation.NOT_EQUAL;
+		}
+
+		if (first == second) { // easiest check of them all!
+			return Relation.EQUAL;
+		}
 
 		Comparator<T1, T2> comparator = getComparator((Class<T1>) first.getClass(), (Class<T2>) second.getClass());
-		if (comparator == null)
+		if (comparator == null) {
 			return Relation.NOT_EQUAL;
+		}
 
 		return comparator.compare(first, second);
 	}
 
 	/**
-	 * A method for obtaining a comparator that can compare two objects of 'firstType' and 'secondType'.
+	 * A method for obtaining a Comparator that can compare two objects of 'firstType' and 'secondType'.
 	 * Please note that comparators may convert objects if necessary for comparisons.
 	 * @param firstType The first type for comparison.
 	 * @param secondType The second type for comparison.
-	 * @return A comparator capable of determine the {@link Relation} between two objects of 'firstType' and 'secondType'.
+	 * @return A Comparator capable of determine the {@link Relation} between two objects of 'firstType' and 'secondType'.
 	 * Will be null if no comparator capable of comparing two objects of 'firstType' and 'secondType' was found.
 	 */
 	@Nullable
-	@SuppressWarnings("unchecked")
 	public static <T1, T2> Comparator<T1, T2> getComparator(Class<T1> firstType, Class<T2> secondType) {
-		if (Skript.isAcceptRegistrations())
+		ComparatorInfo<T1, T2> info = getComparatorInfo(firstType, secondType);
+		return info != null ? info.comparator : null;
+	}
+
+	/**
+	 * A method for obtaining the info of a Comparator that can compare two objects of 'firstType' and 'secondType'.
+	 * Please note that comparators may convert objects if necessary for comparisons.
+	 * @param firstType The first type for comparison.
+	 * @param secondType The second type for comparison.
+	 * @return The info of a Comparator capable of determine the {@link Relation} between two objects of 'firstType' and 'secondType'.
+	 * Will be null if no info for comparing two objects of 'firstType' and 'secondType' was found.
+	 */
+	@Nullable
+	@SuppressWarnings("unchecked")
+	public static <T1, T2> ComparatorInfo<T1, T2> getComparatorInfo(Class<T1> firstType, Class<T2> secondType) {
+		if (Skript.isAcceptRegistrations()) {
 			throw new SkriptAPIException("Comparators cannot be retrieved until Skript has finished registrations.");
+		}
 
 		Pair<Class<?>, Class<?>> pair = new Pair<>(firstType, secondType);
-		Comparator<T1, T2> comparator;
+		ComparatorInfo<T1, T2> comparator;
 
 		if (QUICK_ACCESS_COMPARATORS.containsKey(pair)) {
-			comparator = (Comparator<T1, T2>) QUICK_ACCESS_COMPARATORS.get(pair);
+			comparator = (ComparatorInfo<T1, T2>) QUICK_ACCESS_COMPARATORS.get(pair);
 		} else { // Compute QUICK_ACCESS for provided types
 			comparator = getComparator_i(firstType, secondType);
 			QUICK_ACCESS_COMPARATORS.put(pair, comparator);
```

Comment:
Should we synchronize over `QUICK_ACCESS_COMPARATORS` here?
`+` e.g. no possibility of getComparator_i being called twice with the same args
`-` might slow down this method in multithreaded env cause one call will wait for the other, even with different args

Reply:
Might be worth synchronizing tbh. If we do, it can just be a regular map yeah?